internal/middleware/account.go

```diff
@@ -0,0 +1,33 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+)
+
+func AccountMiddleware(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		rhId := ctxval.GetIdentity(r.Context())
+		accDao, err := dao.GetAccountDao(r.Context())
+		if err != nil {
+			ctxval.GetLogger(r.Context()).Error().Err(err).Msg("Failed to initialize connection to fetch Account info")
+			// TODO should we fail hard?
+			next.ServeHTTP(w, r)
+			return
+		}
+		acc, err := accDao.GetOrCreateByIdentity(r.Context(), rhId.Identity.OrgID, rhId.Identity.AccountNumber)
+		if err != nil {
+			ctxval.GetLogger(r.Context()).Warn().Err(err).Msg("Failed to fetch account by identity")
+			// TODO do something about it
+			next.ServeHTTP(w, r)
+		} else {
+			ctxval.GetLogger(r.Context()).Info().Msgf("Current account: orgId '%s', number: '%s'", acc.OrgID, *acc.AccountNumber)
+			ctx := context.WithValue(r.Context(), ctxval.AccountCtxKey, acc)
+			next.ServeHTTP(w, r.WithContext(ctx))
+		}
```

Comment:
Please keep the line of sight. In other words, drop the else block, it's the success code path otherwise we will eventually end up with 8 inner else blocks.

https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88